src/sqlfluff/core/linter.py

```diff
@@ -33,7 +33,7 @@
 from .parser import Lexer, Parser
 from .string_helpers import findall
 from .templaters import TemplatedFile
-from .rules import get_ruleset
+from sqlfluff.core.rules import get_ruleset
```

Comment:
> Nice approach. Two areas that I think we should do differently.
> 
> * One is that we refine the rule code in the class _and_ the file. This feels like unnecessary duplication, and that we should only define it in one and the other should be generic. Based on the convo in #680 I think we should code the file and so the class in the file should have a generic name.
> * Imports appear inconsistent within the file (some are absolute and some are relative). If this isn't necessary I think we should be consistent and make them relative. If there are some occasions where absolute references are _necessary_ then this should be documented with a comment to say why.

With you on the first - how does this sit with your idea of using subclasses to auto determine rules @barrywhart ? If we went for the route of named file and generic classname, how would we infer the rule name from the subclass?

On the second - my understanding is that absolute imports are preferred over relative. Where I had to adjust/add an import, I went for absolute, but didn't touch the rest. The main reason is that an absolute import is visually far more expressive in terms of project structure, and and it's easier to fix issues too if the project structure moves around with a find and replace vs relative imports. We should definitely be consistent though - I'll make an issue to decide one way or the other, let's make the decision there and take action in another PR?

Reply:
Ah - we inherit a few rules from other rules. If we have generic classnames, that won't work as we'll be redeclaring the class after we've imported it. Any idea how we could overcome that? Or maybe best again to make this decision in another issue? I can see value in having the rule name in the class name too just so it's harder to get confused when working on them, renaming isn't too much of a pain I don't think especially as we're hard-coding the rule names elsewhere (e.g. tests)

Reply:
Also - Sphinx relies on the classname for the documentation generation as another complication!

Reply:
Good points all around. I'm not wedded to the idea of using subclasses, but it seems like it may have some advantages based on the above. We shall see...

My main concern from an initial review was about names and structure being overly prescribed, so Alan's point that the module and class names shouldn't _both_ contain the rule number aligns with my thinking. Based on the above point about Sphinx, I suggest that we allow module names to be arbitrary and get the rule numbers from the class names. 

I suggest we choose module names that are descriptive of the rule, e.g. `trailing_whitespace.py` contains `class Rule_L001`.